shm/player.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <iostream>
+#include <memory>
+#include <vector>
+#include "ship.hpp"
```

Comment:
You can make a forward declaration and include both headers in .cpp file. This causes compilation faster because any changes in both files will not trigger rebuild this one.